### PR TITLE
Document release process

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: docs/dev/README.md [hunk 1]
```diff
@@ -235,3 +235,34 @@ For measuring time of incremental analysis, use either of these:
 $ cargo run --release -p rust-analyzer -- analysis-bench ../chalk/ --highlight ../chalk/chalk-engine/src/logic.rs
 $ cargo run --release -p rust-analyzer -- analysis-bench ../chalk/ --complete ../chalk/chalk-engine/src/logic.rs:94:0
 ```
+
+# Release Process
+
+Release process is handled by `release`, `dist` and `promote` xtasks, `release` being the main one.
+
+`release` assumes that you have checkouts of `rust-analyzer`, `rust-ananalyzer.github.io`, and `rust-lang/rust` in the same directory:
```
Comment (lnicola) [inline docs/dev/README.md:243]: ananananana

Comment (matklad) [reply docs/dev/README.md:243]: hm?

Comment (lnicola) [reply docs/dev/README.md:243]:
```suggestion
`release` assumes that you have checkouts of `rust-analyzer`, `rust-analyzer.github.io`, and `rust-lang/rust` in the same directory:
```

Comment (matklad) [reply docs/dev/README.md:243]: ah.... Now I want to build bananalyzer.

**Comment by jonas-schievink** [comment]
Great!

bors r+

**Pull request merged**